### DISCUSSION TITLE
希望增加LeetCode插件的使用

### DISCUSSION THREAD

**Discussion by yaxproo** [coder/code-server#4851] [Ideas]
增加LeetCode插件的使用，可以随时随地的刷题

**Comment by lg-nineoneninethreenine** [comment]
![what???](https://i.pinimg.com/originals/2f/a7/de/2fa7de205d72b2c48c1ddd95de358136.gif)

FYI, we don't know Chinese, so can you try using Google Translate or other translation engines and tell us what you mean?

**Comment by YueMiyuki** [comment]
He said he hope that code-server include this
https://leetcode.com

**Reply by YueMiyuki** [reply]
@im-coder-lg

**Reply by lg-nineoneninethreenine** [reply]
Oh, an extension! If a VS Code extension exists, or if it is on Open VSX, we could add it here since all you have to do is add it to the registry. All you gotta do is make it, push it to Open VSX and VSCode's marketplace registry and getting it on your instances.

**Reply by lg-nineoneninethreenine** [reply]
What'd he mean of this?
`Increase the use of the LeetCode plugin, you can brush questions anytime, anywhere`
I don't get it.

**Reply by yaxproo** [reply]
> im
There are many topics in leetcode. If codeserver supports it, you can write code on ipad without opening a PC.

**Reply by yaxproo** [reply]
> 哦，一个扩展！如果 VS Code 扩展存在，或者它位于 Open VSX 上，我们可以在此处添加它，因为您所要做的就是将其添加到注册表中。您所要做的就是制作它，将其推送到Open VSX和VSCode的市场注册表，并将其放在您的实例上。

Install leetcode with vsx. Blank when opening question

**Reply by lg-nineoneninethreenine** [reply]
> > im
> There are many topics in leetcode. If codeserver supports it, you can write code on ipad without opening a PC. 
> 

That's literally why code-server even exists! Can you please elaborate?

**Reply by lg-nineoneninethreenine** [reply]
> Install leetcode with vsx. Blank when opening question

Exactly what I told.

**Reply by yaxproo** [reply]
> > Install leetcode with vsx. Blank when opening question
> 
> Exactly what I told.

After leetcode is successfully installed through vsx, open the topic in leetcode, the page is blank

**Reply by lg-nineoneninethreenine** [reply]
Oh it's a bug. Now you're talkin'! Can you share a recording and details of your OS?

**Reply by jsjoeio** [reply]
Hmm..not sure how the leetcode extension works but yeah, if you think there's a bug, please provide repro steps and we can take a look?

**Reply by yaxproo** [reply]
> Oh it's a bug. Now you're talkin'! Can you share a recording and details of your OS?

![动画](https://user-images.githubusercontent.com/51498153/153698429-5f2804e8-1f4b-476d-b9c0-d56e13afdeea.gif)

**Reply by Yac836** [reply]
i have  same problem. when  i use  https , the problem  resolved.
![image](https://github.com/coder/code-server/assets/16701883/e6c959a9-1896-4cb3-b392-90ede7e41f10)

**Comment by lg-nineoneninethreenine** [comment]
These are my guesses:
HTTPS is not enabled
LeetCode's bug
Some files are missing
Can you share a screenshot of DevTools(Ctrl+Shift+I) Console?

**Reply by yaxproo** [reply]
> These are my guesses: HTTPS is not enabled LeetCode's bug Some files are missing Can you share a screenshot of DevTools(Ctrl+Shift+I) Console?

![image](https://user-images.githubusercontent.com/51498153/153698664-3c1600b9-fe2f-4717-a97c-c580f8598057.png)

**Reply by lg-nineoneninethreenine** [reply]
Have you tried disabling Incognito on Chrome? It looks like Chrome so a safe assumption. I think there's a flag in `chrome://flags` to disable Incognito.

**Reply by lg-nineoneninethreenine** [reply]
Kinda feeling like I have seen this error somewhere. Now, try this: Open code-server and the DevTools console. Then, open anything in LeetCode and check if those logs appear.

**Reply by yaxproo** [reply]
> Have you tried disabling Incognito on Chrome? It looks like Chrome so a safe assumption. I think there's a flag in `chrome://flags` to disable Incognito.

Incognito mode is not turned on

**Reply by lg-nineoneninethreenine** [reply]
Alright, you gotta give me more data. What OS are you on, if it's a Docker deployment or local machine or if code-server's initiated on the cloud with the version of code-server and LeetCode you are using. It seems to be WebView, but with more data, I might potentially come up with the bug. Try this template:
```markdown
Local OS: Your OS
Remote OS: (if local OS, fill in Local OS)
code-server version: (You can find it by typing code-server --version)
LeetCode version:
Browser:
```

**Reply by yaxproo** [reply]
> Alright, you gotta give me more data. What OS are you on, if it's a Docker deployment or local machine or if code-server's initiated on the cloud with the version of code-server and LeetCode you are using. It seems to be WebView, but with more data, I might potentially come up with the bug. Try this template:
> 
> ```gfm
> Local OS: Your OS
> Remote OS: (if local OS, fill in Local OS)
> code-server version: (You can find it by typing code-server --version)
> LeetCode version:
> Browser:
> ```

Local OS: windows10 Professional 20H2
Remote OS: CentOS Linux release 7.9.2009 (Core)
code-server version: code-server-4.0.2-linux-amd64
LeetCode version: 0.80.0
Browser: Microsoft Edge 98.0.1108.50

**Reply by lg-nineoneninethreenine** [reply]
My thinking is that MS Edge has some security settings regarding WebView. Have you tried other browsers like Chrome, Vivaldi or Opera? Try Opera, since the former browsers are based on Chromium.

**Reply by yaxproo** [reply]
> My thinking is that MS Edge has some security settings regarding WebView. Have you tried other browsers like Chrome, Vivaldi or Opera? Try Opera, since the former browsers are based on Chromium.

Opera, firefox, chrome can not run

**Reply by lg-nineoneninethreenine** [reply]
Windows in S mode?

**Reply by lg-nineoneninethreenine** [reply]
That's uncool. Try disabling it since you can barely do anything with S mode.

**Reply by yaxproo** [reply]
> That's uncool. Try disabling it since you can barely do anything with S mode.

S mode is not turned on

**Reply by lg-nineoneninethreenine** [reply]
Try sharing a recording of DevTools on Chrome, Firefox, Opera, etc.

**Reply by lg-nineoneninethreenine** [reply]
All of them.

**Reply by yaxproo** [reply]
> Try sharing a recording of DevTools on Chrome, Firefox, Opera, etc.

same error

**Reply by lg-nineoneninethreenine** [reply]
I'm confused. @jsjoeio please help

**Reply by YueMiyuki** [reply]
![image](https://user-images.githubusercontent.com/76854136/153705353-eaa8490c-13de-4b47-a280-702c13f4af88.png)
Hmm

**Reply by YueMiyuki** [reply]
TabNine not working too, by the way

**Reply by lg-nineoneninethreenine** [reply]
Have you tried resetting your VM and trying everything from scratch? It could also be HTTPS.

**Reply by lg-nineoneninethreenine** [reply]
> ![image](https://user-images.githubusercontent.com/76854136/153705353-eaa8490c-13de-4b47-a280-702c13f4af88.png) Hmm

He installed it through the CLI.

**Reply by lg-nineoneninethreenine** [reply]
> TabNine not working too, by the way

Error logs?

**Reply by YueMiyuki** [reply]
> > TabNine not working too, by the way
> 
> Error logs?

Theres no log 
the tabnine hub cant show

**Reply by lg-nineoneninethreenine** [reply]
`code-server --log -debug`?

**Comment by YueMiyuki** [comment]
![image](https://user-images.githubusercontent.com/76854136/153782887-9d32b01d-9078-4767-b9ad-1d55689c6aa2.png)

**Reply by lg-nineoneninethreenine** [reply]
Oh it's `code-server --log debug`, sorry for that :(

**Reply by YueMiyuki** [reply]
![image](https://user-images.githubusercontent.com/76854136/153802669-492aa00b-45c7-403d-8098-ac5cce41dfef.png)
Umm @im-coder-lg

**Reply by lg-nineoneninethreenine** [reply]
I have seen this somewhere, was this fixed?

**Reply by jsjoeio** [reply]
Hmm that sounds like you're trying to open code-server inside VS Code or code-server. Is that the case?

**Reply by lg-nineoneninethreenine** [reply]
That's what I thought!

**Reply by YueMiyuki** [reply]
oh oops

**Reply by YueMiyuki** [reply]
![image](https://user-images.githubusercontent.com/76854136/153987122-fe984c85-278e-4275-bcd9-bd951c517396.png)
I reinstall it and theres no log
